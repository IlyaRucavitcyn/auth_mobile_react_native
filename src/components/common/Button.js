/**@flow */
import * as React from 'react';
import { Text, TouchableOpacity } from 'react-native';

type PropTypes = {
    onPress: any,
    children: React.Node | Array<React.Node>,
    disabled?: boolean
}

const Button = ({ onPress, children, disabled }: PropTypes): React.Node => {
    const { buttonStyle, textStyle } = styles;
    return (
        <TouchableOpacity
            onPress={onPress}
            style={buttonStyle}
            disabled={disabled}
        >
            <Text style={textStyle}>
                {children}
            </Text>
        </TouchableOpacity>
    );
};

const styles = {
    textStyle: {
        alignSelf: 'center',
        color: '#007aff',
        fontSize: 16,
        fontWeight: '600',
        paddingTop: 10,
        paddingBottom: 10
    },
    buttonStyle: {
        flex: 1,
        alignSelf: 'stretch',
        backgroundColor: '#FFF',
        borderRadius: 5,
        borderWidth: 1,
        borderColor: '#007aff',
        marginLeft: 5,
        marginRight: 5
    }
};

export { Button };
