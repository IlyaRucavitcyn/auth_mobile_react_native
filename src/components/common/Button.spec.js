import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import { Button } from './Button';

test('renders snapshot correctly', () => {
    const wrapper = shallow(
        <Button
            disabled={true}
            onPress={() => { }}>
            Testing
        </Button>
    );
    expect(wrapper).toMatchSnapshot();
});

test('inner text is passed correctly', () => {
    const wrapper = shallow(
        <Button>Testing text</Button>
    );
    expect(wrapper.childAt(0).childAt(0).text())
        .toEqual('Testing text');
});

test('disabling attribute is passed correctly', () => {
    const wrapper = shallow(
        <Button disabled={false}>Testing text</Button>
    );
    expect(wrapper.prop('disabled'))
        .toBe(false);
});

test('onPress attribute is passed correctly', () => {
    const wrapper = shallow(
        <Button onPress={() => "testing result"}>Testing text</Button>
    );
    expect(wrapper.prop('onPress')())
        .toBe("testing result");
});
