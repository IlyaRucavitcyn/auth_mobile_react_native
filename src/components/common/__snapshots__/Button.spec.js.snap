// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    disabled={true}
    onPress={[Function]}
  >
    Testing
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.2,
      "children": <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "#007aff",
            "fontSize": 16,
            "fontWeight": "600",
            "paddingBottom": 10,
            "paddingTop": 10,
          }
        }
      >
        Testing
      </Text>,
      "disabled": true,
      "onPress": [Function],
      "style": Object {
        "backgroundColor": "#FFF",
        "borderColor": "#007aff",
        "borderRadius": 5,
        "borderWidth": 1,
        "flex": 1,
        "marginLeft": 5,
        "marginRight": 5,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "accessible": true,
        "allowFontScaling": true,
        "children": "Testing",
        "ellipsizeMode": "tail",
        "style": Object {
          "alignSelf": "center",
          "color": "#007aff",
          "fontSize": 16,
          "fontWeight": "600",
          "paddingBottom": 10,
          "paddingTop": 10,
        },
      },
      "ref": null,
      "rendered": "Testing",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "color": "#007aff",
              "fontSize": 16,
              "fontWeight": "600",
              "paddingBottom": 10,
              "paddingTop": 10,
            }
          }
        >
          Testing
        </Text>,
        "disabled": true,
        "onPress": [Function],
        "style": Object {
          "backgroundColor": "#FFF",
          "borderColor": "#007aff",
          "borderRadius": 5,
          "borderWidth": 1,
          "flex": 1,
          "marginLeft": 5,
          "marginRight": 5,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Testing",
          "ellipsizeMode": "tail",
          "style": Object {
            "alignSelf": "center",
            "color": "#007aff",
            "fontSize": 16,
            "fontWeight": "600",
            "paddingBottom": 10,
            "paddingTop": 10,
          },
        },
        "ref": null,
        "rendered": "Testing",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
